Yosemite/Yosemite/Stores/ReceiptStore.swift

```diff
@@ -217,6 +219,47 @@ private extension ReceiptStore {
             DDLogError("⛔️ Unable to save receipt for order id: \(order.orderID)")
         }
     }
+
+    /// Sends the receipt for the order to the provided email address.
+    /// Updates the billing address of the order to the provided email address and triggers the sending of the receipt.
+    /// - Parameters:
+    ///  - order: The order for which the receipt is being sent.
+    ///  - email: The email address to which the receipt is being sent.
+    ///  - onCompletion: The completion block to call when the operation is complete.
+    ///
+    func sendReceipt(order: Order, email: String, onCompletion: @escaping (Result<Order, Error>) -> Void) {
+        let updatedBillingAddress = order.billingAddress?.copy(email: email) ?? Address(firstName: "",
```

Comment:
non-blocking 💭 : I might be missing the high-level design, if the order already has an email set to the billing address, does it make sense to override the email here?

Reply:
It's a very good question.

If the order already has an email (customer) set, we won't use this method for sending the receipt. I will update the method to reflect this.